src/net/ssl-context.hpp

```diff
@@ -0,0 +1,29 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef SRC_NET_SSL_CONTEXT_HPP
+#define SRC_NET_SSL_CONTEXT_HPP
+
+// Forward declarations
+struct ssl_st;
+struct ssl_ctx_st;
+
+namespace mk {
+namespace net {
+
+class SslContext {
```

Comment:
I don't think you cannot easily get an instance of the class, given that the constructor is private, but I agree that for robustness we should declare it like so! Hold on, cooking the diff...